ddtrace/internal/remoteconfig/client.py

```diff
@@ -33,7 +33,7 @@
     from typing import Tuple
     from typing import Union
 
-    ProductCallback = Callable[[Optional["ConfigMetadata"], Union[Mapping[str, Any], bool, None]], None]
+    ProductCallback = Optional[Callable[[Optional["ConfigMetadata"], Union[Mapping[str, Any], bool, None]], None]]
```

Comment:
Looks reasonable, @avara1986

Reply:
We added Optional because `ProductCallback` is optional in all cases we  return a `ProductCallback` or use it as parameter. It's a semantic debate ;) we could use:

```
ProductCallback = ...
OptionalProductCallback = Optional[ProductCallback]
```